### PR TITLE
Deprecation of `size_ngh` for PR #109

### PR DESCRIPTION
**Pull request by glemaitre**
I just gave an example in what they would do in scikit-learn for the deprecation to take place.
It should be also inserted in the changelog a bit differently

### CODE REVIEW

File: imblearn/combine/smote_enn.py [hunk 1]
```diff
@@ -104,14 +113,16 @@ class SMOTEENN(SamplerMixin):
 
     def __init__(self, ratio='auto', random_state=None,
                  k=5, m=10, out_step=0.5, kind_smote='regular',
-                 n_neighbors=3, kind_enn='all', n_jobs=-1, **kwargs):
+                 size_ngh=3, n_neighbors=3, kind_enn='all', n_jobs=-1,
+                 **kwargs):
```
Comment (chkoar) [inline imblearn/combine/smote_enn.py:116]: **kwargs should be removed?

**Comment by glemaitre** [comment]
yep you right. There should still be some in other module

**Comment by chkoar** [comment]
What if someone provides n_neighbors=5 and size_ngh=4?
If we need both parameters just not to break a user's code, we could just decorate the class with deprecation warning. What do you think?

**Comment by glemaitre** [comment]
You mean as here explained there (section deprecation):
http://scikit-learn.org/stable/developers/contributing.htm

The advantage to have both is that the user can already make the change. If he wants to use both parameters, this is its mistake, you warn him in the documentation and at runtime. We should check what in the repo of scikit-learn. I took the second solution since that it was the first thing that I saw done in the last version of their API.

**Comment by glemaitre** [comment]
https://github.com/scikit-learn/scikit-learn/search?p=1&q=deprecation&utf8=%E2%9C%93

**Comment by chkoar** [comment]
We could inform the user early in the construction of the Sampler instead of fit like here https://github.com/scikit-learn/scikit-learn/blob/b28043281c5bf33c1b17d32becb2035684d93ce0/sklearn/decomposition/nmf.py#L989

A warning is fine in the `__init__` I think. What do you think?

**Comment by glemaitre** [comment]
I fine with that if it pass the test of the base estimator
